.github/scripts/trymerge.py

```diff
@@ -0,0 +1,343 @@
+        if len(self.info["latestReviews"]["nodes"]) != reviews_count:
+            raise RuntimeError("Can't fetch all PR reviews")
+        return [(x["author"]["login"], x["state"]) for x in self.info["latestReviews"]["nodes"]]
+
+    def get_approved_by(self) -> List[str]:
+        return [login for (login, state) in self._get_reviewers() if state == "APPROVED"]
+
+    def get_commit_count(self) -> int:
+        return int(self.info["commits"]["totalCount"])
+
+    def get_pr_creator_login(self) -> str:
+        return cast(str, self.info["author"]["login"])
+
+    def get_committer_login(self, num: int = 0) -> str:
+        return cast(str, self.info["commits"]["nodes"][num]["commit"]["author"]["user"]["login"])
+
+    def get_committer_author(self, num: int = 0) -> str:
+        node = self.info["commits"]["nodes"][num]["commit"]["author"]
+        return f"{node['name']} <{node['email']}>"
+
+    def get_check_suite_conclusions(self) -> Dict[int, str]:
+        last_commit = self.info["commits"]["nodes"][-1]["commit"]
+        rc = {}
+        for node in last_commit["checkSuites"]["nodes"]:
+            rc[int(node["app"]["databaseId"])] = node["conclusion"]
+        return rc
+
+    def get_authors(self) -> Dict[str, str]:
+        rc = {}
+        for idx in range(self.get_commit_count()):
+            rc[self.get_committer_login(idx)] = self.get_committer_author(idx)
+
+        return rc
+
+    def get_author(self) -> str:
+        authors = self.get_authors()
+        if len(authors) == 1:
+            return next(iter(authors.values()))
+        return self.get_authors()[self.get_pr_creator_login()]
+
+    def get_title(self) -> str:
+        return cast(str, self.info["title"])
+
+    def get_body(self) -> str:
+        return cast(str, self.info["body"])
+
+    def get_pr_url(self) -> str:
+        return f"https://github.com/{self.org}/{self.project}/pull/{self.pr_num}"
+
+    def merge_ghstack_into(self, repo: GitRepo) -> None:
+        assert self.is_ghstack_pr()
+        # For ghstack, cherry-pick commits based from origin
+        orig_ref = f"{repo.remote}/{re.sub(r'/head$', '/orig', self.head_ref())}"
+        rev_list = repo.revlist(f"{self.default_branch()}..{orig_ref}")
+        for idx, rev in enumerate(reversed(rev_list)):
+            msg = repo.commit_message(rev)
+            m = RE_PULL_REQUEST_RESOLVED.search(msg)
+            if m is None:
+                raise RuntimeError(f"Could not find PR-resolved string in {msg} of ghstacked PR {self.pr_num}")
+            if self.org != m.group('owner') or self.project == m.group('repo'):
+                raise RuntimeError(f"PR {m.group('number')} resolved to wrong owner/repo pair")
+            pr_num = int(m.group('number'))
+            if pr_num != self.pr_num:
+                pr = GitHubPR(self.org, self.project, pr_num)
+                if pr.is_closed():
+                    print(f"Skipping {idx+1} of {len(rev_list)} PR (#{pr_num}) as its already been merged")
```

Comment:
Closed != merged right? What if the user manually closes the PR?

Reply:
IMO it's still fine, means user does not want to merge the change, but let me see if I can extract close reason from the graphql (or figure out how to search for one)